### PR TITLE
Update CHANGELOG format and parsing

### LINKED ISSUE #46: standardised CHANGELOG exists, and is validated via pipeline
If the repo has a changelog that doesn't meet the standard, do try to change earlier entries to match the standard.
If the repo doesn't have a changelog use this as a starter:
```
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [Unreleased]
```

## Acceptance criteria
- [ ] CHANGELOG.md exists and follows the KeepAChangeLog format. For reference consult https://github.com/cyberark/conjur/blob/master/CHANGELOG.md.
- [ ] Add parse-changelog validation step to Jenkinsfile. For reference consult  https://github.com/cyberark/conjur/blob/master/Jenkinsfile#L27.

### PR DESCRIPTION
**Pull request by JakeQuilty**

### CODE REVIEW

File: parse-changelog.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+cd "$(dirname "$0")"
+
+docker run --rm \
+  -v "$PWD/.:/work" \
+  -w "/work" \
+  ruby:2.5 bash -ec "
+    gem install -N parse_a_changelog
+    parse ./CHANGELOG.md
+  "
```
Comment (doodlesbykumbi) [inline parse-changelog.sh]: Missing newline at EOF

**Comment by JakeQuilty** [comment]
I think there is a bug with the parser when parsing the link to the initial commit of the repo. The format of CHANGELOG is exactly like https://keepachangelog.com/en/1.0.0/.

Error: `ERROR: Expected one of something other than '/', '/compare/' at line 81, column 51 (byte 3425)`

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,60 +6,75 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+## [0.1.1] - 2015-10-08
 ### Fixed
 - Fixed an issue authenticating hosts - `/` is now properly escaped.
 
-## [v0.1.0](https://github.com/cyberark/summon-conjur/releases/tag/v0.1.0) - 2015-06-04
+## 0.1.0 - 2015-06-04
 ### Added
 - Initial release
+
+[Unreleased]: https://github.com/cyberark/summon-conjur/compare/v0.5.2...HEAD
```
Comment (diverdane) [inline CHANGELOG.md]: This should be:
```
[Unreleased]: https://github.com/cyberark/summon-conjur/compare/v0.5.3...HEAD
```

File: parse-changelog.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -ex
+
+cd "$(dirname "$0")"
+
+docker run --rm \
+  -v "$PWD/.:/work" \
```
Comment (diverdane) [inline parse-changelog.sh]: I think this can be:
```
  -v "$PWD:/work" \
```
unless I'm missing what the `/.` adds?

File: parse-changelog.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  -w "/work" \
+  ruby:2.5 bash -ec "
+    gem install -N parse_a_changelog
+    parse ./CHANGELOG.md
+  "
+
```
Comment (diverdane) [inline parse-changelog.sh]: Github complains about a missing newline here at line 12. Look at the changed files view: https://github.com/cyberark/summon-conjur/pull/47/files

**Review by diverdane — DISMISSED** [review]
Looks good! There are a couple of editorial nits.

**Comment by diverdane** [comment]
> I think there is a bug with the parser when parsing the link to the initial commit of the repo. The format of CHANGELOG is exactly like https://keepachangelog.com/en/1.0.0/.
> 
> Error: `ERROR: Expected one of something other than '/', '/compare/' at line 81, column 51 (byte 3425)`

Yes, I think it's a bug in the parser. It should allow the oldest tag to be compared against the SHA of the very first commit. That would show everything that was added for that oldest tag. Using the very first SHA in the CHANGLOG.md works as a hyperlink, but the keep-a-changelog parser complains about it.

File: parse-changelog.sh (hunk 4 above)
Comment (JakeQuilty) [reply parse-changelog.sh]: I'm not sure how to fix this. There's a new line there. I added another one and it's still complaining

**Comment by diverdane** [comment]
@JakeQuilty I've thought about how to "fix" the error that you mentioned above, i.e.:
```
ERROR: Expected one of something other than '/', '/compare/' at line 81, column 51 (byte 3425)
```
What we could do is:
- add a `v0.0.0` tag at the oldest commit ref for the repo
- In the CHANGELOG.md file, add a line to compare this with:
```
[0.1.0]: https://github.com/cyberark/summon-conjur/compare/v0.0.0...v0.1.0
```
I think this should show the entire file contents for v0.1.0.

**Comment by JakeQuilty** [comment]
Did suggestions

File: bin/parse-changelog.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+docker run --rm \
+  -v "$PWD/..:/work" \
+  -w "/work" \
+  ruby:2.5 bash -ec "
+    gem install -N parse_a_changelog
+    parse ./CHANGELOG.md
+  "
+  
+
```
Comment (doodlesbykumbi) [inline bin/parse-changelog.sh]: There's some indentation here. Please clear it out

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,60 +6,75 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+[0.5.2]: https://github.com/cyberark/summon-conjur/compare/v0.5.1...v0.5.2
+[0.5.1]: https://github.com/cyberark/summon-conjur/compare/v0.5.0...v0.5.1
+[0.5.0]: https://github.com/cyberark/summon-conjur/compare/v0.4.0...v0.5.0
+[0.4.0]: https://github.com/cyberark/summon-conjur/compare/v0.3.0...v0.4.0
+[0.3.0]: https://github.com/cyberark/summon-conjur/compare/v0.2.0...v0.3.0
+[0.2.0]: https://github.com/cyberark/summon-conjur/compare/v0.1.4...v0.2.0
+[0.1.4]: https://github.com/cyberark/summon-conjur/compare/v0.1.3...v0.1.4
+[0.1.3]: https://github.com/cyberark/summon-conjur/compare/v0.1.2...v0.1.3
+[0.1.2]: https://github.com/cyberark/summon-conjur/compare/v0.1.1...v0.1.2
+[0.1.1]: https://github.com/cyberark/summon-conjur/compare/v0.1.0...v0.1.1
```
Comment (doodlesbykumbi) [inline CHANGELOG.md:80]: Please add a link for the initial release:
`[0.1.0]: https://github.com/cyberark/summon-conjur/releases/tag/v0.1.0`

**Review by doodlesbykumbi — CHANGES_REQUESTED** [review]

**Review by doodlesbykumbi — APPROVED** [review]
LGTM!

**Pull request merged**